plugins/browser-plugin-accelerator-ecommerce/src/api.ts

```diff
@@ -0,0 +1,228 @@
+      _context[trackerId] = [];
+    },
+    contexts: () => {
+      return _context[trackerId] || [];
+    },
+  };
+}
+
+/**
+ * Track a checkout step
+ *
+ * @param event - The checkout step index and the attributes the user filled in.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackCheckoutStep(
+  event: CheckoutStep & CommonEcommerceEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  const { context = [], timestamp, ...checkoutStep } = event;
+  context.push({ schema: CHECKOUT_STEP_SCHEMA, data: { ...checkoutStep } });
+
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildEcommerceActionEvent({ type: 'checkout_step' }), context, timestamp);
+  });
+}
+
+/**
+ * Track a product list view
+ *
+ * @param event - The list name along with the products which were viewed by the visitor.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackProductListView(
+  event: ListViewEvent & CommonEcommerceEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  const { context = [], timestamp, products = [], name } = event;
+  products.forEach((product) => context.push({ schema: PRODUCT_SCHEMA, data: { ...product } }));
+
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildEcommerceActionEvent({ type: 'list_view', name }), context, timestamp);
+  });
+}
+
+/**
+ * Track a product view/detail
+ *
+ * @param event - The product which was viewed in a product detail page.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackProductView(
+  event: Product & CommonEcommerceEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  const { context = [], timestamp, ...product } = event;
+  context.push({ schema: PRODUCT_SCHEMA, data: { ...product } });
+
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildEcommerceActionEvent({ type: 'product_view' }), context, timestamp);
+  });
+}
+
+/**
+ * Track a product removal from cart
+ *
+ * @param event - The product/s which was/were added to the cart and the total value of the cart after the product was added.
+ * @param trackers - The tracker identifiers which the event will be sent to
+ */
+export function trackAddToCart(
+  event: Cart & CommonEcommerceEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+) {
+  const { context = [], timestamp, products = [], ...cart } = event;
+  products.forEach((product) => context.push({ schema: PRODUCT_SCHEMA, data: { ...product } }));
+  context.push({ schema: CART_SCHEMA, data: { ...cart } });
+
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildEcommerceActionEvent({ type: 'add_to_cart' }), context, timestamp);
+  });
+}
+
+/**
+ * Track a product removal from cart
```

Comment:
This is a bit confusing to me. Does one track the state of the cart after the products were removed or does one track the products to be removed? I assume the second, but using the `Cart` type to represent the `event` leads me to think the first.

Also the comment shouldn't probably talk about a singular product since the `Cart` type only has a `products` list.

I think what would help me would be to put examples in these docstrings. I think that'd actually be a good practice and I have been trying to do that elsewhere. We usually use the Snowplow docs to give examples, but putting them right in the docstrings makes it easier for users to understand the API in IDEs (users love examples). I'm thinking something like:

```suggestion
 * Track removal of products from cart
 *
 * Example:
 * ```
 * trackRemoveFromCart({
 *   products: [
 *      { id: 123 } // product to be removed from the cart
 *   ]
 * });
 * ```
```

Maybe I'm just exaggerating the problem, so just leaving for you to consider.